### PR TITLE
GitHub Actions: Deploy production-release

### PR DESCRIPTION
**Pull request by eatyourgreens**
- add a `deploy_production` workflow, which builds and deploys a production image on pushes to the `production-release` tag.
- delete the Jenkinsfile.

### CODE REVIEW

**Comment by eatyourgreens** [comment]
There's probably a very good reason that we don't re-use the Docker staging build with the same commit hash, but I can't remember what it is.

File: .github/workflows/deploy_production.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - production-release
+  workflow_dispatch:
+
+jobs:
+  build_and_push_image:
+    name: Build Production Image
+    runs-on: ubuntu-latest
+    steps:
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
```
Comment (camallen) [inline .github/workflows/deploy_production.yml:15]: maybe switch to v2 of the login action? 

```suggestion
      uses: docker/login-action@v2
```

File: .github/workflows/deploy_production.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
```
Comment (camallen) [inline .github/workflows/deploy_production.yml:20]: I think you want to use the buildx system in docker (the new one)
```suggestion
    - name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v2

```

**Review by camallen — APPROVED** [review]
I think this is good any. Any errors can be iterated on to fix it.

**Comment by camallen** [comment]
> There's probably a very good reason that we don't re-use the Docker staging build with the same commit hash, but I can't remember what it is.

I think historically the env had different values baked into it at build time so it was hard to pull the same image and configure at boot / runtime to use the prod system values. That may have changed and could be worth iterating on but short term as a jenkins replacement i think this is ok as we isolate the images to their own container namespace in GHCR

File: .github/workflows/deploy_production.yml (hunk 1 above)
Comment (eatyourgreens) [reply .github/workflows/deploy_production.yml:15]: Sure. I'll update #2531 to try this out with the static deploy.

**Comment by eatyourgreens** [comment]
Yeah, I think the webpack production and staging builds, in NextJS, output different JS bundles and we've had problems, in the past, where scripts 404'ed in a production page because they weren't found on a staging image.

File: .github/workflows/deploy_production.yml (hunk 2 above)
Comment (eatyourgreens) [reply .github/workflows/deploy_production.yml:20]: I've started a build on #2531 with BuildX enabled. If that works, I'll add it here too.

**Comment by zwolf** [comment]
Does this action do anything specific to FEM that can't be handled by the shared build/push and deploy_app workflows in ci-cd? If not, why not use them?

**Comment by eatyourgreens** [comment]
I don't know. I copied the `deploy_staging.yml` workflow, but I can take a look at the shared workflows.

**Comment by eatyourgreens** [comment]
If we can use the shared workflows, we should probably update `deploy_branch.yml`, `deploy_staging.yml` and `deploy_production.yml` in a separate PR. They're all pretty much based around the same template.

**Pull request merged**